assignment2/task1.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: UTF-8 -*-
+
+
+class AncestorsFinder(object):
+    def __init__(self, graph):
+        self.graph = graph
+        self.root = 0
+        self.ancestors = None
+
+    def find(self, value):
+        self.ancestors = []
+        self._dfs(self.root, -1, value)
+        return self.ancestors
+
+    def _dfs(self, ver, prev, value):
+        if self.graph[ver]['value'] == value:
```

Comment:
Check current vertex: if it contain value, go out from recursion and save all previous vertexes in  `self.ancestors` attribute during the exit. Previous vertexes - vertexes, from which we can go to current. List `self.ancestors` will be the answer. If current vertex doesn't contain `value`, see comments below